Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_datetimeperiod.py

```diff
@@ -312,88 +312,104 @@ def is_valid_connector_for_date_and_time_period(self, text: str):
 
         return False
 
-    def match_time_of_day(self, source: str, reference: datetime, date_ers: [ExtractResult] = None):
-        tokens: List[Token] = list()
-        source = source.strip().lower()
+    def match_time_of_day(self, source: str, reference: datetime, date_extract_results: [ExtractResult] = None):
+        tokens = []
+        result = []
 
-        matches = regex.finditer(
-            self.config.specific_time_of_day_regex, source)
-        tokens.extend(map(lambda x: Token(x.start(), x.end()), matches))
+        matches = list(regex.finditer(self.config.specific_time_of_day_regex, source))
```

Comment:
Same here

Reply:
Here, finditer return an object and to work with it, we have to parse it into a list, so it is okey as lists in Python work like arrays